build-tools/build-indexes

```diff
@@ -628,6 +628,9 @@ process.stdout.write("Building `data/teambuilder-tables.js`... ");
 	// Learnset table
 	//
 
+	const gen3HMs = {'cut':1, 'fly':1, 'surf':1, 'strength':1, 'flash':1, 'rocksmash':1, 'waterfall':1, 'dive':1};
```

Comment:
We haven't done this in a long time; benchmarking shows arrays and sets are faster and more readable. I'd probably use a `Set` here.

Reply:
+1 for `Set` for readability, even though somewhat counterintutively an array would likely be faster.

Reply:
Ah shoot, yeah. Got too used to that idiom. Since this is a build script, readability is probably more important than speed, so ``Set`` it is.
Wouldn't an Array#includes be O(n) while an object or Set is closer to O(1)? Does the abstraction offset the complexity for these relatively small amounts of data?

Reply:
I have suspicions that there are enough elements here for `Set` to be faster than `Array`, but yes, for small arrays, arrays are faster than sets because of the relative amounts of overhead.

Reply:
Yeah, my intuitions were wrong about how big the `Array` would need to be before `Set` won out - I figured around 5 elements would be the break even point.
https://jsperf.com/pokemon-showdown-includes-vs-has/ (on my monster Linux box, `has` wins, and on my Macbook `includes` wins, and they're pretty close and thats for 10 elements) . Yes. algorithmic complexity is a thing, but in the real world constant factors tend to really matter. I'd never want to discourage using `Set` for something like this though - semantically it makes sense, even if linear search of small arrays beats it.

Reply:
In my experience, in the real world, constant factors stop mattering at around n=15. n=8 is a toss-up, so it makes sense that it'd be different between powerful/unpowerful machine.